### PR TITLE
Initial Implementation

### PR DESCRIPTION
**Pull request by blimmer**
This PR introduces the bulk of the logic for a working asdf plugin for Brig.

Here's it working in the Docker container:

```
> docker run -it --rm -v `pwd`:/root/.asdf/plugins/brig --entrypoint bash asdf-test:dev

root@50d8835f011f:/# asdf list-all brig
v0.2.0
v0.3.0
v0.5.0
v0.6.0
v0.7.0
v0.8.0
v0.9.0
v0.10.0
v0.11.0
v0.12.0
v0.13.0
v0.13.1
v0.14.0
v0.15.0
v0.16.0
v0.17.0
v0.18.0

root@50d8835f011f:/# asdf install brig v0.16.0 && asdf install brig v0.18.0
Downloading brig from https://github.com/Azure/brigade/releases/download/v0.16.0/brig-linux-amd64 to /tmp/brig_GaEGMcq/brig-v0.16.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   607    0   607    0     0   1681      0 --:--:-- --:--:-- --:--:--  1686
100 33.6M  100 33.6M    0     0  2739k      0  0:00:12  0:00:12 --:--:-- 5420k
Creating bin directory
Cleaning previous binaries
Copying binary
Downloading brig from https://github.com/Azure/brigade/releases/download/v0.18.0/brig-linux-amd64 to /tmp/brig_BElbiJJ/brig-v0.18.0
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   607    0   607    0     0   2078      0 --:--:-- --:--:-- --:--:--  2078
100 30.8M  100 30.8M    0     0  1987k      0  0:00:15  0:00:15 --:--:-- 3666k
Creating bin directory
Cleaning previous binaries
Copying binary
root@50d8835f011f:/# asdf list birg
No such plugin: birg
root@50d8835f011f:/# asdf list brig
  v0.16.0
  v0.18.0
root@50d8835f011f:/# asdf global brig v0.16.0
root@50d8835f011f:/# brig version
v0.16.0
```

See inline comments for more detail.

### CODE REVIEW

File: bin/install [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  exit 1
+}
+
+download_url() {
+  local version=$1
+  local platform
+  platform=$(uname -s | tr '[:upper:]' '[:lower:]')
+
+  if [ "$(uname -m)" != 'x86_64' ]; then
+    echo "ERROR: you're running a 32-bit processor. You'll need to find another solution."
```
Comment (blimmer) [inline bin/install:59]: They don't publish pre-built binaries for 32 bit processors so I didn't want to solve this right away.

File: bin/install [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  rm -f "$binary_path" 2>/dev/null || true
+
+  echo "Copying binary"
+  cp "$tmp_download_dir/brig-$version" "$binary_path"
+  chmod +x "$binary_path"
+}
+
+install_ref() {
+  echo "TODO: building from a ref is not yet supported"
+  exit 1
```
Comment (blimmer) [inline bin/install:50]: We can support this with a clone and using their `Makefile`, but it's not MVP in my eyes.

File: bin/install [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  if [ "$(uname -m)" != 'x86_64' ]; then
+    echo "ERROR: you're running a 32-bit processor. You'll need to find another solution."
+    exit 1
+  fi
+
+  echo "https://github.com/Azure/brigade/releases/download/${version}/brig-${platform}-amd64"
+}
+
+installer "$ASDF_INSTALL_TYPE" "$ASDF_INSTALL_VERSION" "$ASDF_INSTALL_PATH"
```
Comment (blimmer) [inline bin/install:66]: See https://github.com/asdf-vm/asdf/blob/ae07c03bd449cab610a0a4c707e38f3ff5f96172/docs/creating-plugins.md for more information on this framework.

**Comment by blimmer** [comment]
The testing framework doesn't seem to like testing locally, so I'll have to flesh those tests out once we open-source this.

**Review by onyxraven — APPROVED** [review]
LGTM for OSS

**Pull request merged**